src/main/java/com/yelp/nrtsearch/server/luceneserver/IndexState.java

```diff
@@ -518,6 +519,8 @@ public IndexState(
     }
     searchThreadPoolExecutor = globalState.getSearchThreadPoolExecutor();
     fetchThreadPoolExecutor = globalState.getFetchService();
+    periodicCommit = new PeriodicCommit(this);
```

Comment:
Can we just set this to null and not create any objects if the `periodicCommitDelaySec` is set to -1?